### DISCUSSION TITLE
Connection of two micro-services (ml-survey-service & assessment-service of  knowledge-inquiry-building block )

### DISCUSSION THREAD

**Discussion by jayesh9747** [Sunbird-inQuiry/Community#108] [Improvements]
i want to connect two micro-service  first is ml-survey-service which build on the node.JS  ,  with the assessment  service. then how to connect ?

**Comment by krgauraw** [comment]
Hi @jayesh9747 the use-case still not clear to me.  assessment service offers api's for Question / QuestionSet. You can use them in ml-survey-service.
If you still have questions, please let me know.

**Reply by jayesh9747** [reply]
@krgauraw  I want to write a migration script to read questions from the assessment service of Sunbird Knowledge Building block. How can I do that? Which APIs of the assessment service provide reading of questions from the database?

**Reply by jayesh9747** [reply]
i want to implement this features in the ml-survey-service 
-  Develop Migration script for transferring Questionnaires from Survey Service to Sunbird Knowledge Building Block.
-  Migration script includes a review process, marking all Questions as published upon reviewer approval.
-  Once Questions are published, execute Migration script for Solutions, updating their status accordingly.
-  All migrated data must be trackable, generating a CSV report containing all relevant statuses.
-  Survey service works with Sunbird Knowlg BB

**Reply by krgauraw** [reply]
Hi @jayesh9747 the assessment-service is not under Knowlg. It is available under inQuiry. 

The service offers Question/ QuestionSet read api's. It also has hierarchy read api. You can use those api's to get the data.
For more details on api's, please [visit](https://inquiry.sunbird.org/learn/product-and-developer-guide/question-and-question-set-service/apis)

Hope this helps!

cc: @sajeshkayyath @PoojaJManjooran

**Reply by jayesh9747** [reply]
@krgauraw  @sajeshkayyath  @PoojaJManjooran  I have thoroughly gone through the ml-survey service documentation and assessment service of inquiry building block documentation.  Now I have concerns regarding how to connect these two different microservices using migration script. Also, could you please provide me with technical overview or documentations  how different microservices are connected in sunbird ?

**Reply by krgauraw** [reply]
Hi @jayesh9747 You need to use api's offered by assessment service only in your migration script. I don't have any information related to ml-survey service.  In sunbird also different micro-services are connected using api only.
If you can share the design document of your migration script, we may be able to help you on specific points. 
cc: @sajeshkayyath @PoojaJManjooran

**Reply by jayesh9747** [reply]
@kameshbhr sure sir, this are the documentation of ml-survey-service 
https://ed.sunbird.org/use/source-code/manage-learn/ml-survey-service
ML-survey service currently takes surveys and observations from students or similar entities. I want to extend this service so that questions can be read from inquiry services. If you are comfortable, can we schedule a small G Meet to discuss this further?

**Reply by krgauraw** [reply]
Hi @jayesh9747 Looks like no design is done as you have shared the microsite link. Sure, you can schedule a call. 
cc: @PoojaJManjooran @sajeshkayyath

**Reply by jayesh9747** [reply]
@krgauraw  can you share your mail id so that i invite for g-meet?

**Reply by jayesh9747** [reply]
@krgauraw  this is my mail id jayeshsavaliya0167@gmail.com . can you please send the invite link for a call on g-meet ?

**Reply by jayesh9747** [reply]
@krgauraw  @PoojaJManjooran  @sajeshkayyath  
here is the sunbird survey service schema for the observation  and survey 
[ML-Survey.pdf](https://github.com/Sunbird-inQuiry/Community/files/15072251/ML-Survey.pdf)

like i want to confused that that in question set schema there are two file like schema.js and config.json then how to move this data of questionaries according two this schema .

**Reply by krgauraw** [reply]
Hi @jayesh9747 you can schedule a call on 24th at 4:00 PM. My email id is: gaurawk@ilimi.in

**Reply by jayesh9747** [reply]
@krgauraw   
I wanted to extend my sincerest apologies for the inconvenience caused by the incorrect timing of the meeting. 
To rectify this, I have created a new meeting with the correct timing. I kindly ask for your understanding and acceptance of the new meeting invitation.

**Comment by jayesh9747** [comment]
@krgauraw, could you clarify whether the Bearer Token required for creating a question and question set needs to be obtained from an administrator? Is this token the same for all users or different for each user? Additionally, how can one obtain this Bearer Token?
![image](https://github.com/Sunbird-inQuiry/Community/assets/123955234/e79c6139-a589-4004-93ba-5ce57cdf43e0)

I want to migrate ML Survey service questions. Is there any deployed Inquiry Building Block available for developers?

**Reply by jayesh9747** [reply]
@krgauraw  please provide guidance and can you share the postman collection of inquiry building block?
